File: languages/cpp/01-foundations/types-and-io/example/input-validation-loop.cpp
#include <iostream>
#include <limits>
#include <string>

using namespace std;

int main() {
    string fullName;
    int age = 0;

    cout << "Enter your full name: ";
    getline(cin, fullName);

    while (true) {
        cout << "Enter your age (1-120): ";
        if (!(cin >> age)) {
            cout << "Invalid input. Please enter a number.\n";
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
            continue;
        }

        if (age < 1 || age > 120) {
            cout << "Age must be between 1 and 120.\n";
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
            continue;
        }

        cin.ignore(numeric_limits<streamsize>::max(), '\n');
        break;
    }

    cout << "\nRegistration summary:\n";
    cout << "Name: " << fullName << '\n';
    cout << "Age: " << age << '\n';

    return 0;
}

File: languages/python/01-foundations/types-and-io/example/input-validation-loop.py
# This extra example extends types-and-io with a validation loop.
# Example purpose: show how string input and numeric checks work together.

full_name = input("Enter your full name: ").strip()

while True:
    raw_age = input("Enter your age (1-120): ").strip()
    try:
        age = int(raw_age)
    except ValueError:
        print("Invalid input. Please enter a number.")
        continue

    if age < 1 or age > 120:
        print("Age must be between 1 and 120.")
        continue

    break

print("\nRegistration summary:")
print(f"Name: {full_name}")
print(f"Age: {age}")
